### PR TITLE
OHRM5X-1428: Recruitment User Role Related Bug Fixes

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR Contains the Following Bug Fixes:

  - OHRM5X-1406
  - OHRM5X-1542
  - OHRM5X-1539
  - OHRM5X-1561
  - OHRM5X-1584
  - OHRM5X-1612
  - OHRM5X-1548
  - OHRM5X-1547
  - OHRM5X-1600

### CODE REVIEW

File: src/client/src/orangehrmRecruitmentPlugin/components/VacancyDropdown.vue [hunk 1]
```diff
@@ -37,6 +37,11 @@ export default {
       required: false,
       default: null,
     },
+    excludeInterviewers: {
```
Comment (Super-Chama) [inline src/client/src/orangehrmRecruitmentPlugin/components/VacancyDropdown.vue:40]: not clear what this is? this is vacancy dropdown, whats the relation with Interviewers?

File: src/client/src/core/util/validation/rules.ts [hunk 2]
```diff
@@ -120,6 +120,14 @@ export const digitsOnly = function(value: string): boolean | string {
   );
 };
 
+export const numericOnly = function(value: string): boolean | string {
```
Comment (Super-Chama) [inline src/client/src/core/util/validation/rules.ts:123]: this function looks same as `digitsOnly()` ? why not update it directly

Comment (SandeepaOHRM) [reply src/client/src/core/util/validation/rules.ts:123]: yes, need to confirm whether we are going to update the digitsOnly with numericOnly or using a separate rule

File: src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        return new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::FILTER_NAME_OR_ID,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::LENGTH, [null, self::PARAM_RULE_FILTER_NAME_OR_ID_MAX_LENGTH]),
+                )
+            ),
+        );
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php:80]: handle pagination and sorting params

File: src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME_OR_ID
+            )
+        );
+        $employees = $this->getVacancyService()->getVacancyDao()->getEmployeeList($employeeListParamHolder);
+        return new EndpointCollectionResult(
+            EmployeeListModel::class,
+            $employees,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => count($employees)])
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php:63]: fix the count

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmRecruitmentPlugin/Dto/EmployeeListSearchFilterParams.php [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Recruitment\Dto;
+
+use OrangeHRM\Core\Dto\FilterParams;
+
+class EmployeeListSearchFilterParams extends FilterParams
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Dto/EmployeeListSearchFilterParams.php:24]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        $employeeListParamHolder = new EmployeeListSearchFilterParams();
+        $this->setSortingAndPaginationParams($employeeListParamHolder);
+
+        $employeeListParamHolder->setNameOrId(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::FILTER_NAME_OR_ID
+            )
+        );
+        $employees = $this->getVacancyService()->getVacancyDao()->getEmployeeList($employeeListParamHolder);
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php:59]: done

File: src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php (hunk 4 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/EmployeeListAPI.php:63]: fixed

File: src/plugins/orangehrmRecruitmentPlugin/Api/Model/EmployeeListModel.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Recruitment\Api\Model;
+
+use OrangeHRM\Core\Api\V2\Serializer\ModelTrait;
+use OrangeHRM\Core\Api\V2\Serializer\Normalizable;
+use OrangeHRM\Entity\Employee;
+
+class EmployeeListModel implements Normalizable
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmRecruitmentPlugin/Api/Model/EmployeeListModel.php:26]: done

**Pull request merged**